src/WAV.jl

```diff
@@ -248,25 +248,40 @@ end
 
 ieee_float_container_type(nbits) = (nbits == 32 ? Float32 : (nbits == 64 ? Float64 : error("$nbits bits is not supported for WAVE_FORMAT_IEEE_FLOAT.")))
 
+## ignore skip for pipelines
+function Base.skip(::Base.Process, Uint64) end
+function Base.skip(::Base.ProcessChain, Uint64) end
+
+function read_raw_sample(io::IO, len::Integer)
+    ispipe = isa(io, Union{Base.Process, Base.ProcessChain})
+    if ispipe
+        return read(io)
+    else
+        return read(io, len)
```

Comment:
Why did this call switch from `read!` to `read`?

Reply:
In the original code, the length of the data to be read was known, and could be declared in advance.  In the case of reading from a pipe, this won't work as way too too much memory is reserved if the process writing to the pipe has lied about the length (as is the case for `sox`).  

I further changed the `read!` to `read` in the non-pipe reading case.  I assumed that it does not make a difference in efficiency where the memory is declared, since the implementation of `read(::IO, len::Int)` declares the memory just like in the original code, and then does the `read!()`.

Reply:
Alright.

Am I correct that this code depends on the other process being well behaved?